DEVELOPER.md

```diff
@@ -0,0 +1,167 @@
+
+1. On your host machine, generate a keypair for SSH into the guest: `ssh-keygen -t ed25519`.
+Just keep hitting Enter until the key is generated. You do not need a password for this key file since it is only used for SSH into your guest
+1. Type `cat .ssh/id_ed25519.pub` and copy the public key
+1. SSH into the guest using command above
+1. Create the ssh config directory (if it doesn't exist) `$ mkdir -p .ssh`
+1. Edit the authorized keys list: `vim .ssh/authorized_keys`
+1. Paste in the content of .ssh/id_ed25519.pub
+1. Adjust the required privileges: `chmod 700 .ssh/`  and `chmod 400 .ssh/authorized_keys`
+1. Logout of guest and make sure you are not promoted password when SSH again
+1. Edit the .ssh/config file on your host and put in the following content:
+
+```
+    Host dev
+        HostName 127.1
+        Port 22222
+        User <your_user_name>
+```
+
+Now try ssh dev on your host, you should be able to get into the guest directly
+
+## Linux Environment
+
+Once you have a Linux development environment (either virtual or bare metal), the build is done in four separate steps:
+
+1. Prerequisite packages.  Mostly compilerls, tools and libraries needed to compile everything else.
+1. OpenResty system, including Nginx, LuaJIT, PCRE, etc.
+1. Databases. Kong uses Posgres, Cassandra and Redis.  We have a handy setup with docker-compose to keep each on its container.
+1. Kong itself.
+
+
+### Prerequisites
+
+These are the needed tools and libraries that aren't installed out of the box on Ubuntu and Fedora, respectively.  Just run one of these, either as root or sudo.
+
+Ubuntu:
+
+```
+    apt-get update
+
+    apt-get install \
+        build-essential \
+        curl \
+        docker \
+        docker-compose \
+        git \
+        libpcre3 \
+        libyaml-dev \
+        m4 \
+        openssl \
+        perl \
+        procps \
+        unzip \
+        zlib1g-dev
+```
+
+Fedora:
+
+```
+    dnf install \
+        automake \
+        docker \
+        docker-compose \
+        gcc \
+        gcc-c++ \
+        git \
+        libyaml-devel \
+        make \
+        patch \
+        pcre-devel \
+        unzip \
+        zlib-devel
+```
+
+### OpenResty
+
+We have a build script that makes it easy to pull and compile specific versions of the needed components of the OpenResty system.  Currently these include OpenResty 1.15.8.1, OpenSSl 1.1.1c, LuaRocks 3.1.3 and PCRE 8.43
+
+These commands don't have to be performed as root, since all compilation is done within a subdirectory, and installs everything in the target specified by the `-p` argument (here the `build` directory).
+
+```
+    git clone https://github.com/kong/openresty-build-tools
+
+    cd openresty-build-tools
+
+    ./kong-ngx-build -p build \
+        --openresty 1.15.8.1 \
+        --openssl 1.1.1c \
+        --luarocks 3.1.3 \
+        --pcre 8.43
+```
```

Comment:
can we put a note here that the necessary prerequisites can be found in `.requirements`?

Reply:
done.